src/components/MagicCodeInput.tsx

```diff
@@ -102,55 +85,70 @@ const decomposeString = (value, length) => {
 /**
  * Converts an array of strings into a single string. If there are undefined or
  * empty values, it will replace them with a space.
- *
- * @param {Array} value
- * @returns {String}
  */
-const composeToString = (value) => _.map(value, (v) => (v === undefined || v === '' ? CONST.MAGIC_CODE_EMPTY_CHAR : v)).join('');
-
-const getInputPlaceholderSlots = (length) => Array.from(Array(length).keys());
-
-function MagicCodeInput(props) {
+const composeToString = (value: string[]): string => value.map((v) => (!v ? CONST.MAGIC_CODE_EMPTY_CHAR : v)).join('');
```

Comment:
suggestion, wouldn't it be better to write `v ?? CONST.MAGIC_CODE_EMPTY_CHAR`

Reply:
Done!